fastq/singleCell.go

```diff
@@ -0,0 +1,65 @@
+package fastq
+
+import (
+	"fmt"
+	"github.com/vertgenlab/gonomics/dna"
+	"github.com/vertgenlab/gonomics/fileio"
+)
+
+//SingleCellPair is a struct that contains single-cell sequencing information, including a pairedEnd read, barcode, and UMI.
+type SingleCellPair struct {
+	Reads PairedEnd
+	Bx    []dna.Base
+	Umi   []dna.Base
+}
+
+//ReadToChanSingleCellPair takes two files, corresponding to R1 and R2, and returns a channel of SingleCellPair structs. It parses the barcode and UMI to user-specified lengths.
+func ReadToChanSingleCellPair(fileOne string, fileTwo string, barcodeLength int, umiLength int, tenXg chan<- SingleCellPair) {
+	fwd, rev := fileio.EasyOpen(fileOne), fileio.EasyOpen(fileTwo)
+	defer fwd.Close()
+	defer rev.Close()
```

Comment:
I would think about closing these files before the "close(tenXg)".  Closing the channel signals to other go routines that this function is done, but the way it is written now it still needs to close two files after that channel is closed.